lib/consumer_supervisor.ex

```diff
@@ -262,6 +258,23 @@ defmodule ConsumerSupervisor do
     call(supervisor, :count_children)
   end
 
+  @doc """
+  Receives a template to initialize and a set of options.
+
+  This is typically invoked at the end of the init/1 callback of module-based supervisors.
+
+  This function returns a the child specification and the supervisor flags.
+
+  """
+  def init([{_, _, _, _, _, _} = template], opts) do
+    {:ok, {_, spec}} = Supervisor.Spec.supervise([template], opts)
+    {:ok, spec, opts}
+  end
+  def init([template], opts) do
+    {:ok, {_, spec}} = Supervisor.init([template], opts)
```

Comment:
`Supervisor.init/2` in Elixir v1.5.1 is capable of handling both formats, so I would implement this `init/1` like this:

```elixir
def init(template, opts) when is_tuple(template) or is_map(template) or is_atom(template) do
  {:ok, {_, [template]}} = Supervisor.init([template], strategy: :one_for_one)
  {:ok, template, opts}
end
```

And let's the consumer supervisor to always expect a single template.